unikernel.ml

```diff
@@ -2948,11 +2860,16 @@ struct
             check_meth `GET (fun () ->
                 authenticate store reqd
                   (email_verification (verify_email store)))
-        | path when String.starts_with ~prefix:"/auth/verify/token=" path ->
+        | "/auth/verify" ->
             check_meth `GET (fun () ->
-                let token = String.sub path 19 (String.length path - 19) in
-                authenticate store reqd
-                  (email_verification (verify_email_token store token)))
+                match get_query_parameter "token" with
+                | Ok token ->
+                    authenticate store reqd
+                      (email_verification (verify_email_token store token))
+                | _ ->
```

Comment:
a case matching everything else (`_ ->`) -> this is code smell and should be removed.

previously, the code was pretty lean, now it exploded in size. could there be a `get_query_parameter` function that errors directly (i.e. sends an error back) if the requested one is not found?

Anyways, maybe you could open an issue about it, and this can be addressed in a different PR?